### PR TITLE
Adding a timeout for each call of istioctl in the test

### PR DESCRIPTION
**Pull request by ingwonsong**
Regarding https://github.com/istio/istio/issues/42448,

Adding a timeout for each call of istioctl from TestXdsProxyStatus test case.

### CODE REVIEW

File: tests/integration/pilot/istioctl_test.go [hunk 1]
```diff
@@ -455,7 +456,7 @@ func TestXdsProxyStatus(t *testing.T) {
 
 			retry.UntilSuccessOrFail(t, func() error {
 				args := []string{
-					"proxy-status", fmt.Sprintf("%s.%s", podID, apps.Namespace.Name()),
+					"x", "proxy-status", fmt.Sprintf("%s.%s", podID, apps.Namespace.Name()), timeoutFlag,
```
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go:459]: oof, does this mean we were no testing it this whole time?

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
This LGTM but just to be clear - https://github.com/istio/istio/issues/42448 is still a real world bug, not just a test flake

File: tests/integration/pilot/istioctl_test.go (hunk 1 above)
Comment (ingwonsong) [reply tests/integration/pilot/istioctl_test.go:459]: not sure...

Comment (ingwonsong) [reply tests/integration/pilot/istioctl_test.go:459]: But when looking at some logs, weirdly "experimental" version of "proxy-status" seems to run in the recent tests. I suspect that somehow "PREFER-EXPERIMENTAL" was turned on in the test. https://github.com/istio/istio/blob/7bf52db38c91c89a4d56d6a94f402fd9ddaf6465/istioctl/cmd/root.go#L198
(previously, this part did not run with "experimental" version of "proxy-status")

**Pull request merged**